src/python/pants/backend/python/rules/download_pex_bin.py

```diff
@@ -77,6 +77,9 @@ def create_process(  # type: ignore[override]
 
         env = dict(env) if env else {}
         env.update(**pex_build_environment.invocation_environment_dict,)
+        if "--pex-root" in pex_args:
```

Comment:
This only affects one of the subclasses of `HermeticPex`... should the other be updated as well? Or should this be implemented in `HermeticPex`?

Reply:
This is the correct spot. Every other HermeticPex should, as an optimization for remote execution environments with unwritable home dir only, use ./pex_root. Since that's not super important, this change is good to go.

Reply:
Hm... the cache directory is always located inside the workspace, so unless we're disabling the cache in the other locations, it should be equivalent. But ok.

Reply:
^ I'm still unclear on the above, but we can followup to fix if need be.

Reply:
To clarify (to myself as well):  

If we set the PEX_ROOT on every other HermeticPex, we must set it to something that *isn't* the named cache that pex.pex uses (and which we now set with --pex-root) - that was exactly the problem in the first place. 

We might want to set it to a *different* named cache, say, `.cache/runtime_pex_root`, so that A) local runs can enjoy the precompiled .pyc files, and B) remote runs don't attempt the unwriteable homedir.